### ISSUE TITLE
Product Attribute Variations out of order on product page. Set up correctly in back end.

### ISSUE THREAD

**Issue by lukerandall1101** [woocommerce/woocommerce#26444]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. No matter what i do, ordered attributes and variations correctly, set to a custom ordering. So the order is S, M , L , XL (very simple)
2. No matter what this will not order correctly on the product page, its just a random jumble
3. I have checked all forums all of which tell me to edit the order in the back end all of which has been completed. I have decided this must be a bug after following the identical advice of every page that it should be working and others chiming in that its happening to them also.

## Expected/actual behavior

When I follow those steps, I see...

I was expecting to see...

## Isolating the problem

- [ x] This bug happens with only WooCommerce plugin active
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin here.
```
</details>
![Screen Shot 2020-05-10 at 11 23 44](https://user-images.githubusercontent.com/53874057/81496968-5b7de500-92b3-11ea-87ad-5344389264e0.png)
![Screen Shot 2020-05-10 at 11 23 08](https://user-images.githubusercontent.com/53874057/81496970-5f116c00-92b3-11ea-87b4-be916dc5f986.png)

**Comment by biser-yordanov** [comment]
Same here

**Comment by negapo** [comment]
Hi lukerandall1101,
WC will not follow the order of the variations but will follow the order in which the attribute is set in the attributes page. Go to the "Attributes" page, look for your attribute "Size", click to configure the attribute and them move up and down the options as you like it to appear on the front end.

**Comment by peterfabian** [comment]
Hi, please have a look at our docs for product taxonomies: 
https://docs.woocommerce.com/document/managing-product-taxonomies/#section-8

TL;DR is that you sort attributes in Dashboard > Products > Attributes (1)
Then under Edit (2), you select custom sort order
And finally, under Configure terms (3) you sort them as you wish by grabbing them by ☰.
<img width="1912" alt="Screenshot 2020-05-11 at 12 11 02" src="https://user-images.githubusercontent.com/2207451/81550586-c1846e00-9380-11ea-93c2-140672f8e22e.png">
<img width="1912" alt="Screenshot 2020-05-11 at 12 13 59" src="https://user-images.githubusercontent.com/2207451/81550765-fc86a180-9380-11ea-8595-faad7a59d941.png">

Those are then displayed in the desired order on the front end (unless you use other extensions that don't respect this order):
<img width="1912" alt="Screenshot 2020-05-11 at 12 17 50" src="https://user-images.githubusercontent.com/2207451/81551049-774fbc80-9381-11ea-8467-e3edf990fa52.png">

**Comment by lukerandall1101** [comment]
Thanks a lot! This is working!
best Wishes
